azavea/echo-locator
# Feature/kak/account search#21

## Overview

Adds an unstyled page at `/select` to search for, select, create, and delete account profiles for voucher holders. Redirect to the page on site load if no profile set. Store profile JSON in browser localStorage and in a file on S3.

Also adds a pull request template and updates the `mastarm` build tools version in the lock file.


## Demo

Search with no parameters returns all:
![image](https://user-images.githubusercontent.com/960264/53591431-f7fe1080-3b61-11e9-8ec3-89b164412b50.png)

On creating a new profile:
![image](https://user-images.githubusercontent.com/960264/53591477-19f79300-3b62-11e9-863e-1f2514b00130.png)

Navigates to map page on profile selection:
![image](https://user-images.githubusercontent.com/960264/53591388-de5cc900-3b61-11e9-8d48-14e48a41a8fb.png)


## Notes

After this, #22 should only require implementing the form and adding the page; the data management parts should be in place.

Currently, any authenticated AWS user can access the profile files; #51 is to restrict it to the Cognito-associated IAM role.

The wireframe for the select page doesn't offer a way to create a new profile, so I added a button to the search form for that. There's also not a delete option in the wireframes, so I made that a list item button. We may want to revisit where those buttons go and how the profile selection flow works.


## Testing

 - `cd taui && yarn start`
 - After login, should see 'select account' page
 - Clicking 'search' with empty fields should show all
 - Should be able to create an account (after entering any text into the two fields)
 - Should see an error message on attempting to create an account with either field empty
 - Should be able to search by fields
 - Should be able to delete account via list button
 - Select an account
 - Should redirect to map page
 - Should show selected voucher holder name and number in the header
 - Header info should persist through router page changes and page refresh
 - Logging out and back in with an account already selected should go to map page directly
 - On map page, delete account profile from localStorage (key `echolocator-profile`)
 - Refresh page; should redirect to select account with no account in header info


Closes #21
